web_monitoring/internetarchive.py

```diff
@@ -141,6 +143,12 @@ def search_cdx(params):
         except Exception:
             raise UnexpectedResponseFormat(text)
 
+        clean_url = REDUNDANT_HTTPS_PORT.sub(
+            r'\1\2', REDUNDANT_HTTP_PORT.sub(
+                r'\1\2', data.url))
+        if clean_url != data.url:
+            data = data._replace(url=clean_url)
```

Comment:
IA will frequently wind up with URLs that have redundant port information in them. Not sure what in their implementation leads to this, but it seemed sensible to clean these up.

There is a related issue of sometimes having both `http:` and `https:` versions of an otherwise identical URL and content, but I thought I’d hold off on trying to think of a fix for that since there’s much less clarity on what’s correct there.